dm-manager/pkg/grpc/grpc_server/grpc_server.go

```diff
@@ -0,0 +1,260 @@
+}
+
+func (dms *DeviceManagementService) RetrieveActivationDetails(ctx context.Context, req *pb.ActivationRequest) (*pb.ActivationDetailsResponse, error) {
+
+	zlog.Info().Msgf("RetrieveActivationDetails")
+
+	if dms.authEnabled {
+		if !dms.rbac.IsRequestAuthorized(ctx, rbac.CreateKey) {
+			err := inv_errors.Errorfc(codes.PermissionDenied, "Request is blocked by RBAC")
+			zlog.InfraSec().InfraErr(err).Msgf("Request Device management is not authenticated")
+		}
+	}
+
+	tenantID, present := inv_tenant.GetTenantIDFromContext(ctx)
+	if !present {
+		err := inv_errors.Errorfc(codes.Unauthenticated, "Tenant ID is missing from context")
+		zlog.InfraSec().InfraErr(err).Msgf("Request Nodes is not authenticated")
+		return nil, err
+	}
+	zlog.Debug().Msgf("ReportAMTStatus: tenantID=%s", tenantID)
+
+	var hostInv *computev1.HostResource
+	var response *pb.ActivationDetailsResponse
+	hostInv, err := dms.invClient.GetHostByUUID(ctx, tenantID, req.HostId)
+	if err != nil {
+		zlog.InfraSec().InfraErr(err).Msgf("Failed to get host by UUID %s", req.HostId)
+		if inv_errors.IsNotFound(err) {
+			zlog.Debug().Msgf("Node Doesn't Exist for UUID %s and tID=%s\n",
+				req.HostId, tenantID)
+			return nil, inv_errors.Errorfc(codes.NotFound, "Host with UUID %s not found", req.HostId)
+		}
+	}
+	zlog.Debug().Msgf("DesiredAmtState %s ", hostInv.DesiredAmtState.String())
+	zlog.Debug().Msgf("CurrentAmtState %s ", hostInv.CurrentAmtState.String())
+
+	if hostInv.DesiredAmtState == computev1.AmtState_AMT_STATE_PROVISIONED && (hostInv.CurrentAmtState == computev1.AmtState_AMT_STATE_UNPROVISIONED ||
+		hostInv.CurrentAmtState == computev1.AmtState_AMT_STATE_UNSPECIFIED) {
+		zlog.Debug().Msgf("Send activation request for Host %s ", req.HostId)
+		response = &pb.ActivationDetailsResponse{
+			Operation:   pb.OperationType_ACTIVATE,
+			HostId:      req.HostId,
+			ProfileName: tenantID,
+		}
+	} else {
+		zlog.Debug().Msgf("Node is provisioned for UUID %s and tID=%s\n",
+			req.HostId, tenantID)
+		return nil, inv_errors.Errorfc(codes.FailedPrecondition, "current state is %s or activation not requested by user %v", hostInv.CurrentAmtState, err)
+	}
+
+	return response, nil
+}
+
+func (dms *DeviceManagementService) ReportActivationResults(ctx context.Context, req *pb.ActivationResultRequest) (*pb.ActivationResultResponse, error) {
+
+	zlog.Info().Msgf("ReportActivationResults")
+
+	if dms.authEnabled {
+		if !dms.rbac.IsRequestAuthorized(ctx, rbac.CreateKey) {
+			err := inv_errors.Errorfc(codes.PermissionDenied, "Request is blocked by RBAC")
+			zlog.InfraSec().InfraErr(err).Msgf("Request Device management is not authenticated")
+		}
+	}
+
+	tenantID, present := inv_tenant.GetTenantIDFromContext(ctx)
+	if !present {
+		err := inv_errors.Errorfc(codes.Unauthenticated, "Tenant ID is missing from context")
+		zlog.InfraSec().InfraErr(err).Msgf("Request Nodes is not authenticated")
+		return nil, err
+	}
+
+	var hostInv *computev1.HostResource
```

Comment:
This variable setting is not needed, hostInv can be directly set from the response from `dms.invClient.GetHostByUUID`